res/controllers/Denon-HC1000S-scripts.js

```diff
@@ -0,0 +1,920 @@
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+DenonHC1000S.shiftSampleLeft = false;
+DenonHC1000S.shiftSampleLeftButton = function(channel, control, value, status, group) {
+    DenonHC1000S.shiftSampleLeft = ! DenonHC1000S.shiftSampleLeft; // '!' inverts a boolean (true/false) value
+    print ("shiftSampleLeft is pressed " + DenonHC1000S.shiftSampleLeft);
+    DenonHC1000S.LED_HotCue_and_Sampler(channel, control, value, status, group);
+    //DenonHC1000S.LED_Play_and_Cue(channel, control, value, status, group);
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight && DenonHC1000S.shift) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
+        print ("want to show Samplers in UI; samplersShow is " + samplersShow);
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+DenonHC1000S.shiftSampleRight = false;
+DenonHC1000S.shiftSampleRightButton = function(channel, control, value, status, group) {
+    DenonHC1000S.shiftSampleRight = ! DenonHC1000S.shiftSampleRight; // '!' inverts a boolean (true/false) value
+    print ("shiftSampleRight is pressed " + DenonHC1000S.shiftSampleRight);
+    DenonHC1000S.LED_HotCue_and_Sampler(channel, control, value, status, group);
+    //DenonHC1000S.LED_Play_and_Cue(channel, control, value, status, group);
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight && DenonHC1000S.shift) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
+        print ("want to show Samplers in UI; samplersShow is " + samplersShow);
+        engine.setValue("[Samplers]", "show_samplers", !samplersShow);
+    }
+};
+
+
+///////
+
+DenonHC1000S.getValue = function(key) {
+    return engine.getValue(DenonHC1000S.group, key);
+};
+
+DenonHC1000S.setValue = function(key, value) {
+    engine.setValue(DenonHC1000S.group, key, value);
+};
+
+DenonHC1000S.samplesPerBeat = function(group) {
+    return 2 * engine.getValue(group, "track_samplerate") * 60 / engine.getValue(group, "file_bpm");
+};
+
+//==TrackSelect===================================================================
+
+DenonHC1000S.recvTrackSelectKnob = function(_channel, _control, value, _status) {
+    var knobDelta = DenonHC1000S.getKnobDelta(value);
+    if (DenonHC1000S.shift) {
+        var ctrlName = "ScrollVertical";
+    } else {
+        var ctrlName = "MoveVertical";
+    }
+    engine.setValue("[Library]", ctrlName, knobDelta);
+};
+
+DenonHC1000S.recvTrackSelectButton = function(_channel, _control, value, _status) {
+    var buttonPressed = DenonHC1000S.isButtonPressed(value);
+    if (!buttonPressed) {
+        return;
+    }
+    if ((DenonHC1000S.shiftSampleLeft && DenonHC1000S.shiftSampleRight) && (value === 64)) {
+        var samplersShow = engine.getValue("[Samplers]", "show_samplers");
```

Comment:
205:13  error    'samplersShow' is assigned a value but never used                          no-unused-vars

This is obvious, remove this.

Reply:
thanks, Holzhaus, I will double check, but it actually is used as I flip the value and use it for debug print  e.g. 
engine.setValue("[Samplers]", "show_samplers", !samplersShow);

Reply:
oh I understand - I use it elsewhere but not here, ok will remove. cheers!